unixfs/examples/cat.rs

```diff
@@ -0,0 +1,159 @@
+use cid::Cid;
+use ipfs_unixfs::file::{visit::IdleFileVisit, FileReadFailed};
+use std::convert::TryFrom;
+use std::fmt;
+use std::io::{Error as IoError, Read, Write};
+use std::path::PathBuf;
+
+fn main() {
+    let cid = match std::env::args().skip(1).next().map(Cid::try_from) {
+        Some(Ok(cid)) => cid,
+        Some(Err(e)) => {
+            eprintln!("Invalid cid given as argument: {}", e);
+            std::process::exit(1);
+        }
+        None => {
+            eprintln!("USAGE: {} CID\n", std::env::args().next().unwrap());
+            eprintln!(
+                "Will walk the unixfs file pointed out by the CID from default go-ipfs 0.5 \
+                configuration flatfs blockstore and write all content to stdout."
+            );
+            std::process::exit(0);
+        }
+    };
+
+    let ipfs_path = match std::env::var("IPFS_PATH") {
+        Ok(s) => s,
+        Err(e) => {
+            eprintln!("IPFS_PATH is not set or could not be read: {}", e);
+            std::process::exit(1);
+        }
+    };
+
+    let mut blocks = PathBuf::from(ipfs_path);
+    blocks.push("blocks");
+
+    let blockstore = ShardedBlockStore { root: blocks };
+
+    match walk(blockstore, &cid) {
+        Ok((read, content)) => {
+            eprintln!("Content bytes: {}", content);
+            eprintln!("Total bytes:   {}", read);
+        }
+        Err(Error::OpeningFailed(e)) => {
+            eprintln!("{}\n", e);
+            eprintln!("This is likely caused by either:");
+            eprintln!(" - ipfs does not have the block");
+            eprintln!(" - ipfs is configured to use non-flatfs storage");
+            eprintln!(" - ipfs is configured to use flatfs with different sharding");
+            std::process::exit(1);
+        }
+        Err(e) => {
+            eprintln!("Failed to walk the merkle tree: {}", e);
+            std::process::exit(1);
+        }
+    }
+}
+
+fn walk(blocks: ShardedBlockStore, start: &Cid) -> Result<(u64, u64), Error> {
+    let stdout = std::io::stdout();
+    let mut stdout = stdout.lock();
+
+    let mut read_bytes = 0;
+    let mut content_bytes = 0;
+
+    // The blockstore specific way of
```

Comment:
I think the comment got accidentally cut here

Reply:
Thank you for trying to read the comments :)